action.yml

```diff
@@ -0,0 +1,104 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: SSH to Google Cloud Platform compute instances
+author: Google LLC
+description: |-
+  Use this action to connect to Google Cloud Platform instances via ssh and execute user's commands.
+
+inputs:
+  instance_name:
+    description: |-
+      Name of the virtual machine instance to SSH into.
+    required: true
+
+  zone:
+    description: |-
+      Zone of the instance to connect to.
+    required: true
+
+  user:
+    description: |-
+      Specifies the username with which to SSH. 
+      If omitted, the user login name is used. If using OS Login, USER will be replaced by the OS Login user.
+    required: false
+
+  ssh_private_key:
+    description: |-
+      SSH private key with which to SSH.
+    required: true
+
+  ssh_public_key:
```

Comment:
Why do we need a public key?

Reply:
If not - we need to retrieve it somehow, it's not clear. Potentially we could make an API call to do that.
As for now as prefer to set it explicitly, probably after alpha version we could update the logic.

Reply:
```node
const pubKeyObject = crypto.createPublicKey({
  key: privateKey,
  format: 'pem'
})

const publicKey = pubKeyObject.export({
  format: 'pem',
  type: 'spki'
})
```